### PR TITLE
docs: Docstring update 

### PR DESCRIPTION
**Pull request by vblagoje**
- fixes https://github.com/deepset-ai/haystack/issues/7133

### CODE REVIEW

**Comment by vblagoje** [comment]
@julian-risch I asked @silvanocerza to review because he's the most familiar with these files

File: integrations/elasticsearch/src/haystack_integrations/components/retrievers/elasticsearch/bm25_retriever.py [hunk 1]
```diff
@@ -11,8 +11,9 @@
 @component
 class ElasticsearchBM25Retriever:
     """
-    ElasticsearchBM25Retriever is a keyword-based retriever that uses BM25 to find the most
-    similar documents to a user's query.
+    ElasticsearchBM25Retriever retrieves documents from the ElasticsearchDocumentStore using BM25 algorithm to find the
+    most similar documents to a user's query.
+
```
Comment (julian-risch) [inline integrations/elasticsearch/src/haystack_integrations/components/retrievers/elasticsearch/bm25_retriever.py:16]: The usage example uses `print(doc.text), which should be `print(doc.content)` instead.

File: integrations/elasticsearch/src/haystack_integrations/components/retrievers/elasticsearch/bm25_retriever.py [hunk 2]
```diff
@@ -96,7 +111,8 @@ def run(self, query: str, filters: Optional[Dict[str, Any]] = None, top_k: Optio
         :param query: String to search in Documents' text.
         :param filters: Filters applied to the retrieved Documents.
         :param top_k: Maximum number of Documents to return.
-        :return: List of Documents that match the query.
+        :return: A dictionary with the following keys:
```
Comment (julian-risch) [inline integrations/elasticsearch/src/haystack_integrations/components/retrievers/elasticsearch/bm25_retriever.py:114]: Let's use `returns` for consistency

File: integrations/elasticsearch/src/haystack_integrations/components/retrievers/elasticsearch/embedding_retriever.py [hunk 3]
```diff
@@ -11,9 +11,30 @@
...
+        Document(text="My name is Carla and I live in Berlin"),
+        Document(text="My name is Paul and I live in New York"),
+        Document(text="My name is Silvano and I live in Matera"),
+        Document(text="My name is Usagi Tsukino and I live in Tokyo"),
+    ]
+    document_store.write_documents(documents)
+
+    result = retriever.run(query="Who lives in Berlin?")
+    for doc in result["documents"]:
+        print(doc.text)
```
Comment (julian-risch) [inline integrations/elasticsearch/src/haystack_integrations/components/retrievers/elasticsearch/embedding_retriever.py:36]: Should be `doc.content`

File: integrations/elasticsearch/src/haystack_integrations/components/retrievers/elasticsearch/embedding_retriever.py [hunk 4]
```diff
@@ -11,9 +11,30 @@
...
+    # Add documents to DocumentStore
+    documents = [
+        Document(text="My name is Carla and I live in Berlin"),
+        Document(text="My name is Paul and I live in New York"),
+        Document(text="My name is Silvano and I live in Matera"),
+        Document(text="My name is Usagi Tsukino and I live in Tokyo"),
+    ]
+    document_store.write_documents(documents)
+
+    result = retriever.run(query="Who lives in Berlin?")
```
Comment (julian-risch) [inline integrations/elasticsearch/src/haystack_integrations/components/retrievers/elasticsearch/embedding_retriever.py:34]: EmbeddingRetriever requires an embedding as input so we need to use a text embedder here.

File: integrations/elasticsearch/src/haystack_integrations/document_stores/elasticsearch/document_store.py [hunk 5]
```diff
@@ -128,11 +134,20 @@ def to_dict(self) -> Dict[str, Any]:
...
+            Dictionary to deserialize from.
+        :returns:
+              Deserialized component.
+        """
         return default_from_dict(cls, data)
 
     def count_documents(self) -> int:
         """
         Returns how many documents are present in the document store.
+        :return: Number of documents in the document store.
```
Comment (julian-risch) [inline integrations/elasticsearch/src/haystack_integrations/document_stores/elasticsearch/document_store.py:150]: :returns:

File: integrations/elasticsearch/src/haystack_integrations/document_stores/elasticsearch/document_store.py [hunk 6]
```diff
@@ -165,6 +180,14 @@ def _search_documents(self, **kwargs) -> List[Document]:
         return documents
 
     def filter_documents(self, filters: Optional[Dict[str, Any]] = None) -> List[Document]:
+        """
+        The main query method for the document store. It retrieves all documents that match the filters.
+
+        :param filters: A dictionary of filters to apply. For more information on the structure of the filters,
+            see the official Elasticsearch documentation:
+            https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl.html
```
Comment (julian-risch) [inline integrations/elasticsearch/src/haystack_integrations/document_stores/elasticsearch/document_store.py:188]: Let's use an anchor text for the link.

File: integrations/elasticsearch/src/haystack_integrations/document_stores/elasticsearch/document_store.py [hunk 7]
```diff
@@ -175,8 +198,13 @@ def filter_documents(self, filters: Optional[Dict[str, Any]] = None) -> List[Doc
     def write_documents(self, documents: List[Document], policy: DuplicatePolicy = DuplicatePolicy.NONE) -> int:
         """
         Writes Documents to Elasticsearch.
+
         If policy is not specified or set to DuplicatePolicy.NONE, it will raise an exception if a document with the
         same ID already exists in the document store.
+
+        :param documents: List of Documents to write to the document store.
+        :param policy: DuplicatePolicy to apply when a document with the same ID already exists in the document store.
+        :return: Number of documents written to the document store.
```
Comment (julian-risch) [inline integrations/elasticsearch/src/haystack_integrations/document_stores/elasticsearch/document_store.py:207]: `:returns:`

File: integrations/elasticsearch/src/haystack_integrations/document_stores/elasticsearch/document_store.py [hunk 8]
```diff
@@ -229,6 +257,8 @@ def _deserialize_document(self, hit: Dict[str, Any]) -> Document:
         """
         Creates a Document from the search hit provided.
         This is mostly useful in self.filter_documents().
+
+        :param hit: A search hit from Elasticsearch.
```
Comment (julian-risch) [inline integrations/elasticsearch/src/haystack_integrations/document_stores/elasticsearch/document_store.py:261]: `:returns:` is missing

**Review by julian-risch — CHANGES_REQUESTED** [review]
Found a couple of smaller things we need to change. In addition to the inline comments,
`__init__` docstring shouldn't mention defaults, for example we should remove `defaults to False
links should use anchor text: 
```
see the official documentation for valid values:
https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#fuzziness
```
should be instead
```
see the [official documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#fuzziness) for valid values.
```

**Review by julian-risch — APPROVED** [review]
LGTM! 👍

**Pull request merged**